### PR TITLE
Create CSS Modules migration post

### PR DESCRIPTION
**Pull request by rebeccadee**
PREVIEW: https://deploy-preview-5159--sourcegraph.netlify.app/blog/migrating-to-css-modules-with-codemods-and-code-insights/

https://app.asana.com/0/1199959077348208/1201686011425862/f

### CODE REVIEW

File: blogposts/2022/css-modules-migration.md
Comment (rebeccadee) [inline blogposts/2022/css-modules-migration.md:141]: Same here, is this right?
image7

File: blogposts/2022/css-modules-migration.md
Comment (rebeccadee) [inline blogposts/2022/css-modules-migration.md:165]: Is this correct?
image4

File: blogposts/2022/css-modules-migration.md
Comment (rebeccadee) [inline blogposts/2022/css-modules-migration.md:171]: Was trying to capture this, need to check if correct
image1

File: blogposts/2022/css-modules-migration.md
Comment (rebeccadee) [inline blogposts/2022/css-modules-migration.md:80]: @valerybugakov We talked about briefly explaining the difference between codemods and Batch Changes but I don't see that anywhere. Can you help add it please?

**Review by rebeccadee — COMMENTED** [review]
@valerybugakov Here's the PR, left a few comments for you please 🤗 You will see we didn't end up with a strong metaphor for the illustration – given the short time frame it was proving too complicated so went with something simple 😁 I see you're out today so I'm going to ask your team if anyone can double check that I've captured the code snippets correctly so there's less for you to look at tomorrow 🙏

File: blogposts/2022/css-modules-migration.md
Comment (rebeccadee) [inline blogposts/2022/css-modules-migration.md:187]: TODO: Fix alignment

File: blogposts/2022/css-modules-migration.md
Comment (rebeccadee) [inline blogposts/2022/css-modules-migration.md:59]: TODO: Add link

File: blogposts/2022/css-modules-migration.md
Comment (rebeccadee) [inline blogposts/2022/css-modules-migration.md:64]: TODO: Add link to live insight

File: blogposts/2022/css-modules-migration.md
Comment (umpox) [inline blogposts/2022/css-modules-migration.md:141]:
```suggestion
```js
const classMapping = {
  'insights-dashboard': 'insights-dashboard',
  'insights-dashboard__wrapper': 'wrapper',
}
```
```

File: blogposts/2022/css-modules-migration.md
Comment (rebeccadee) [reply blogposts/2022/css-modules-migration.md:171]:
```suggestion
`$ yarn transform --write --transform ./globalCssToCssModule.ts ./sourcegraph/**/*.tsx`
```

File: blogposts/2022/css-modules-migration.md
Comment (umpox) [reply blogposts/2022/css-modules-migration.md:132]: LGTM!

File: blogposts/2022/css-modules-migration.md
Comment (umpox) [inline blogposts/2022/css-modules-migration.md:159]:
```suggestion
```jsx
<div className="insights-dashboard__wrapper" />
```
```

File: blogposts/2022/css-modules-migration.md
Comment (umpox) [inline blogposts/2022/css-modules-migration.md:165]:
```jsx
<div className={styles.wrapper} />
```
```suggestion
```jsx
<div className={styles.wrapper} />
```
```

File: blogposts/2022/css-modules-migration.md
Comment (umpox) [inline blogposts/2022/css-modules-migration.md:171]:
```suggestion
```sh
$ yarn transform --write --transform ./globalCssModule.ts ./sourcegraph/**/*.tsx
```
```

**Review by umpox — COMMENTED** [review]
Left some comments on the code examples

File: blogposts/2022/css-modules-migration.md
Comment (rebeccadee) [reply blogposts/2022/css-modules-migration.md]: Applied in batched suggestions

File: blogposts/2022/css-modules-migration.md
Comment (rebeccadee) [reply blogposts/2022/css-modules-migration.md:64]: @umpox @valerybugakov Do you have a link I can use here? We wanted to embed it but that's not possible on the blog yet.

(Also, just noticed the alt text is wrong, I will fix that)

File: blogposts/2022/css-modules-migration.md
Comment (rebeccadee) [reply blogposts/2022/css-modules-migration.md:59]:
```suggestion
Code Insights entered Beta in August 2021, and we happily started using it to track the migration progress. As of today, [Code Insights is now Generally Available](/blog/announcing-code-insights/).
```

File: blogposts/2022/css-modules-migration.md
Comment (umpox) [reply blogposts/2022/css-modules-migration.md:64]: A link directly to the code insight? We don't unfortunately, this insight isn't publicly accessible and AFAIK we can't directly link the code insights right now

File: blogposts/2022/css-modules-migration.md
Comment (valerybugakov) [reply blogposts/2022/css-modules-migration.md]: Fixed

**Review by valerybugakov — APPROVED** [review]
Head up @rebeccadee: I pushed cosmetic improvements for this post. Please check them out on the preview. Let me know if you think something needs to be reverted. 

Since we have more time because of the delayed launch, I will find time this week to address the `TODO` left in the code.

File: blogposts/2022/css-modules-migration.md
Comment (valerybugakov) [reply blogposts/2022/css-modules-migration.md:92]: Hey @rebeccadee, since then, I've updated this paragraph with a detailed description of what codemods do. We do not reference Batch Changes in the blog post, and this name is specific to the Sourcegraph product; I feel it's redundant to raise this topic here. What do you think?

File: blogposts/2022/css-modules-migration.md
Comment (valerybugakov) [reply blogposts/2022/css-modules-migration.md]: Confirmed with Code Insights team. It's not possible to reference or embed this insight.

File: blogposts/2022/css-modules-migration.md
Comment (rebeccadee) [reply blogposts/2022/css-modules-migration.md:92]: @valerybugakov Agreed, thank you!

File: blogposts/2022/css-modules-migration.md
Comment (rebeccadee) [inline blogposts/2022/css-modules-migration.md:35]:
```suggestion
We researched popular solutions available in the open source universe: trade-offs between CSS in JS, regular CSS, and hybrid solutions. Some offered more powerful features, but we decided to use a solution that we knew we could easily adopt—CSS modules.
```

File: blogposts/2022/css-modules-migration.md
Comment (rebeccadee) [inline blogposts/2022/css-modules-migration.md:43]:
```suggestion
With CSS modules, CSS classes should be referenced in the JavaScript file via explicit binding to the styles file:
```
This might not be quite right but we're definitely missing a word here! @valerybugakov

**Review by rebeccadee — COMMENTED** [review]
@valerybugakov Thank you so much for fixing up the styling and resolving the TODO 🤗 One question for you!

File: blogposts/2022/css-modules-migration.md
Comment (rebeccadee) [inline blogposts/2022/css-modules-migration.md:5]:
```suggestion
publishDate: 2022-03-10T00:00-07:00
```

**Comment by felixfbecker** [comment]
Heads up, we need an approving review by tomorrow morning before we can merge this PR.

**Comment by rebeccadee** [comment]
@valerybugakov Would you be able to give an approving review so I can merge later? 🙏

**Review by valerybugakov — APPROVED** [review]
👩‍🚀

**Pull request merged**
